### ISSUE TITLE
梯度裁剪的问题

### ISSUE THREAD

**Issue by ZL-BUAA** [wenzhu23333/Differential-Privacy-Based-Federated-Learning#7]
你好，我看了你的代码，per_sample_clip这个函数，是裁剪每一层而不是单样本裁剪的意思么？

**Comment by ZL-BUAA** [comment]
loss.backward()反向传播已经执行了，再执行clip_gradients()，此时裁剪的是模型的每一层的权重吧，如何单样本梯度联系起来

**Comment by wenzhu23333** [comment]
本仓库使用了Opacus库Wrap了原有模型，使得net会有额外的一个sample_gradient属性，其维度与batch size大小相同，存储了每条sample的梯度。
参考：[https://github.com/wenzhu23333/Differential-Privacy-Based-Federated-Learning/blob/master/README_CN.md#%E9%99%84%E6%B3%A8](中文ReadMe)
以及 [https://opacus.ai/tutorials/guide_to_grad_sampler](Opacus GradSampleModule)
